utils/src/main/java/software/amazon/awssdk/utils/ToStringResult.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.utils;
+
+import java.util.Arrays;
+import software.amazon.awssdk.annotations.NotThreadSafe;
+
+/**
+ * A class to standardize implementations of {@link Object#toString()} across the SDK.
+ *
+ * <pre>{@code
+ * ToStringResult.builder("Person")
+ *               .add("name", name)
+ *               .add("age", age)
+ *               .build();
+ * }</pre>
+ */
+@NotThreadSafe
+public final class ToStringResult {
+    private final StringBuilder result;
+    private final int startingLength;
+
+    /**
+     * @see #builder(String)
+     */
+    private ToStringResult(String className) {
+        this.result = new StringBuilder(className).append("{");
+        this.startingLength = result.length();
+    }
+
+    /**
+     * Create a to-string result for the given class name.
+     */
+    public static String create(String className) {
+        return className + "{}";
```

Comment:
The manual `toStrings` we were doing before were usually "ClassName()" rather than "ClassName{}" - why the switch? I think we should go back to roundy brackets, especially when there's parameters it seems more natural to me to have: `ClassName(member=value, array=[value1, value2])`

Reply:
We used braces in our modeled objects (eg. `GetObjectRequest`), so I had to pick one to go with. Braces or square brackets seemed more standard in other toString implementations like Apache's ToStringBuilder, guava's ToStringHelper and IntelliJ's toString template, so I didn't want to go against the grain without a good reason.

I don't mind switching if everyone can agree on one.